### PR TITLE
add short README.md blurb about specification

### PR DESCRIPTION
**Pull request by jadekler**
This is a short message intended to orient folks new to this repository
about the layout of tests, how protos and verifiers fit in, and how
tests are actually used.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,3 +4,25 @@ This repository is used to host a collection of conformance tests
...
+The shape of the test per-client is largely up to the implementor. Similarly,
+whether the .json files are hand-written or generated by some other program
+is up to the implementor.
+
+Conformance test runners must be implemented per language, per client (for
+clients that have conformance tests defined here). These runners must do the
+minimal work to translate a conformance test specified here into a testcase
+that inputs the given input data and asserts based on the given assertion.
+
+Generally, conformance tests are accessed as git submodules in the repositories.
```
Comment (frankyn) [inline README.md]: Link to git submodules.

File: README.md [hunk 2]
```diff
@@ -4,3 +4,25 @@ This repository is used to host a collection of conformance tests
...
 
+## Conformance Test Specification
+
+Conformance tests in these repos generally share these characteristics:
+
+- Tests are specified as a set of inputs and expected outputs, and are defined
+in .json files.
+- Test .json files are paired with a .proto that defines the specification for
+the .json file. The .json files are verified to match the .proto by a simple
+validator program that's run in CI.
```
Comment (frankyn) [inline README.md]: Please add a link to the storage/v1 proto and json files.

File: README.md [hunk 3]
```diff
@@ -4,3 +4,25 @@ This repository is used to host a collection of conformance tests
...
+
+The shape of the test per-client is largely up to the implementor. Similarly,
+whether the .json files are hand-written or generated by some other program
+is up to the implementor.
```
Comment (frankyn) [inline README.md]: Add "For example, storage/v1 json file is hand written for ease of update."

File: README.md [hunk 4]
```diff
@@ -4,3 +4,25 @@ This repository is used to host a collection of conformance tests
...
+
+Conformance test runners must be implemented per language, per client (for
+clients that have conformance tests defined here). These runners must do the
+minimal work to translate a conformance test specified here into a testcase
+that inputs the given input data and asserts based on the given assertion.
```
Comment (frankyn) [inline README.md]: I'm confused by this paragraph. Did you mean when conformance tests must be used by each language client to verify its implementation and the client verify must be written in the client's language?

File: README.md (hunk 2 above)
Comment (jadekler) [reply README.md]: As an example? (ex: [storage .proto](link), [storage .proto validator](link)).

File: README.md (hunk 3 above)
Comment (jadekler) [reply README.md]: Hmm... do you think we need such an example? It seems self-explanatory: pointing out that storage/v1 is hand-written doesn't seem to add to the point.

I don't have fresh eyes, though, so I'd defer to you if you're sure.

File: README.md (hunk 4 above)
Comment (jadekler) [reply README.md]: > Did you mean [...] conformance tests must be used by each language client to verify its implementation
Yes

File: README.md (hunk 1 above)
Comment (jadekler) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (frankyn) [reply README.md]: +1, just to not have a person click around for examples.

File: README.md (hunk 4 above)
Comment (frankyn) [reply README.md]: sg, at first I read that the conformance tests validator implemented in this repo must be implemented in all languages.

File: README.md (hunk 3 above)
Comment (frankyn) [reply README.md]: I was thinking about being explicit about existing decisions. Let's not include this change given links to example proto and json file in v1/storage above.

Comment (jadekler) [reply README.md]: Ack

File: README.md (hunk 2 above)
Comment (jadekler) [reply README.md]: Done

**Comment by jadekler** [comment]
(all comments addressed)

**Pull request merged**